### DISCUSSION TITLE
Division by zero in btVoronoiSimplexSolver::closestPtPointTriangle

### DISCUSSION THREAD

**Discussion by fredUmlaut** [bulletphysics/bullet3#3503] [General]
There are various places in this function where a division by zero does occur in rare cases.

Line 387:
if (vb <= btScalar(0.0) && d2 >= btScalar(0.0) && d6 <= btScalar(0.0)) {
  btScalar w = d2 / (d2 - d6);   //Division by Zero if d2==d6

A similar issue is in line 399 and perhaps line 410.

**Comment by erwincoumans** [comment]
thanks for the report, I'll look at it. Dou you have a reproduction case you can share?

**Comment by fredUmlaut** [comment]
It happens in three separate commercial games but i can't provide those as test case, sorry.
